src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,392 @@
+                    + "              ::::::::::::::::::.      .         `::::'" + "\n"
+                    + "              ::::::::::::::::::::,     .          ::" + "\n"
+                    + "               ::::::::::::::::::::      :         ::" + "\n"
+                    + "                  ::::::::::::::::'      .       .::'" + "\n"
+                    + "                  :: '::::::::::::'       .     .:':" + "\n"
+                    + "                   ::    ~~::::''        . ,,,::::,,:::::" + "\n"
+                    + "                   :::,,,,,,,,,,,....:::::::::::::::::::::" + "\n"
+                    + "                  :::::::::::::::::   :::::::::::::::::::::" + "\n"
+                    + "                 .:::::::::::::::::,,  :::::::::::::::::''" + "\n"
+                    + "                .::::::::::::::::::::,   :::::::::::::'" + "\n"
+                    + "                `:::::::::::::::::::::,'" + "\n";
+
+
+    /** Constants used to set status of a task */
+    private static final String DEFAULT_STATUS = " ";
+    private static final String DONE_STATUS = "X";
+
+
+    /** Invalid input messages */
+    private static final String DEFAULT_INVALID_MESSAGE
+            = "I'm sorry, I don't quite understand :( Could you try again?";
+    private static final String MISSING_FIELDS_MESSAGE
+            = "I think you missed some fields! Try again?";
+    private static final String INVALID_INDEX_MESSAGE
+            = "Squeal! Second field must be a number.";
+    private static final String OUTSIDE_RANGE_INDEX_MESSAGE
+            = "Squeal? There is no task in the list with index ";
+    private static final String INVALID_TASK_MESSAGE
+            = "Squeal... Are you sure that is a task?";
+
+
+    /** List of tasks being maintained and number of tasks it has */
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static int tasksCount = 0;
+
+
+    /** Methods that display messages */
+    public static void greet() {
+        String greetString = BORDER + NEWLINE
+                + LOGO + NEWLINE
+                + "Hello, I'm Panda!" + NEWLINE
+                + "What would you like to do today?" + NEWLINE
+                + "Tip: use \"help\" to view all valid commands" + NEWLINE
+                + BORDER + NEWLINE + NEWLINE;
+        System.out.print(greetString);
+    }
+
+    public static void goodbye() {
+        String goodbyeString = BORDER + NEWLINE
+                + "Alright! Goodbye :)" + NEWLINE
+                + BORDER + NEWLINE;
+        System.out.print(goodbyeString);
+    }
+
+    public static void printInvalidInputMessage(String message) {
+        String openString = BORDER + NEWLINE;
+        String closeString = NEWLINE + BORDER + NEWLINE + NEWLINE;
+        if (message != null) {
+            System.out.print(openString + message + closeString);
+            return;
+        }
+        System.out.print(openString + DEFAULT_INVALID_MESSAGE + closeString);
```

Comment:
Good use of guard clause! Although in this case guarding `message == null` since printing the message can be thought of as the happy path in this case.